File: bindings/java-jni/src/main/java/org/maplibre/nativejni/camera/CameraFitOptions.java
package org.maplibre.nativejni.camera;

import java.util.Objects;

/** Mutable descriptor for camera fitting queries. */
public final class CameraFitOptions {
  private EdgeInsets padding;
  private Double bearing;
  private Double pitch;

  public boolean hasPadding() {
    return padding != null;
  }

  public EdgeInsets padding() {
    return padding;
  }

  public CameraFitOptions padding(EdgeInsets padding) {
    this.padding = Objects.requireNonNull(padding, "padding");
    return this;
  }

  public CameraFitOptions clearPadding() {
    padding = null;
    return this;
  }

  public boolean hasBearing() {
    return bearing != null;
  }

  public Double bearing() {
    return bearing;
  }

  public CameraFitOptions bearing(double bearing) {
    this.bearing = bearing;
    return this;
  }

  public CameraFitOptions clearBearing() {
    bearing = null;
    return this;
  }

  public boolean hasPitch() {
    return pitch != null;
  }

  public Double pitch() {
    return pitch;
  }

  public CameraFitOptions pitch(double pitch) {
    this.pitch = pitch;
    return this;
  }

  public CameraFitOptions clearPitch() {
    pitch = null;
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/camera/CameraFitOptions.kt
package org.maplibre.nativeffi.camera

/** Mutable camera fitting descriptor. */
public class CameraFitOptions {
  public var padding: EdgeInsets? = null
    private set

  public var bearing: Double? = null
    private set

  public var pitch: Double? = null
    private set

  public fun hasPadding(): Boolean = padding != null

  public fun padding(padding: EdgeInsets): CameraFitOptions = apply { this.padding = padding }

  public fun clearPadding(): CameraFitOptions = apply { padding = null }

  public fun hasBearing(): Boolean = bearing != null

  public fun bearing(bearing: Double): CameraFitOptions = apply { this.bearing = bearing }

  public fun clearBearing(): CameraFitOptions = apply { bearing = null }

  public fun hasPitch(): Boolean = pitch != null

  public fun pitch(pitch: Double): CameraFitOptions = apply { this.pitch = pitch }

  public fun clearPitch(): CameraFitOptions = apply { pitch = null }
}
